lib/phoenix_live_view.ex

```diff
@@ -498,7 +496,60 @@ defmodule Phoenix.LiveView do
                   "got: #{inspect(other)}"
       end
 
-    %{container: container, name: name, kind: :view, module: module, layout: layout}
+    phoenix_live_mount = Module.get_attribute(env.module, :phoenix_live_mount)
+    lifecycle = Phoenix.LiveView.Lifecycle.mount(env.module, phoenix_live_mount)
+
+    namespace =
+      opts[:namespace] || env.module |> Module.split() |> Enum.take(1) |> Module.concat()
+
+    name = env.module |> Atom.to_string() |> String.replace_prefix("#{namespace}.", "")
+    container = opts[:container] || {:div, []}
+
+    live = %{
+      container: container,
+      name: name,
+      kind: :view,
+      module: env.module,
+      layout: layout,
+      lifecycle: lifecycle
+    }
+
+    quote do
+      @doc false
+      def __live__ do
+        unquote(Macro.escape(live))
+      end
+    end
+  end
+
+  @doc """
+  Declares a hook to be attached to the `:mount` stage of the LiveView lifecycle.
+
+  The `on_mount/1` macro provides an injection point for hooks to run
+  before both the disconnected _and_ connected render of the LiveView.
+  Hooks declared `on_mount` are invoked immediately before the `c:mount/3`
+  callback. A hook that wishes to redirect the LiveView during the
+  `:mount` stage **must** halt, otherwise an error will be raised.
+
+  For more information about lifecycle hooks, see `attach_hook/4`.
+
+  ## Examples
+
+      defmodule DemoWeb.PageLive do
+        use Phoenix.LiveView
+
+        on_mount {DemoWeb.LiveAuth, :ensure_mounted_current_user}
+        on_mount DemoWeb.InitAssigns
+      end
```

Comment:
Please provide examples of how the actual callback would look like in this case. :)